charts/prometheus-mysql-exporter/templates/serviceaccount.yaml

```diff
@@ -8,8 +8,16 @@ metadata:
     chart: {{ template "prometheus-mysql-exporter.chart" . }}
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
-  {{- with .Values.serviceAccount.annotations }}
+  {{- if .Values.cloudsqlproxy.workloadIdentity.enabled }}
+  annotations:
+  {{- with .Values.serviceAccount.annotations  }}
+      {{- toYaml . | nindent 4 }}
+  {{- end }}
+    {{ template "prometheus-mysql-exporter.workloadIdentityAnnotation" . }}
+  {{- else }}
+  {{- with .Values.serviceAccount.annotations  }}
   annotations:
```

Comment:
now we might have annotations twice in the file.
please fix.

Reply:
Hi @monotek, thanks for reviewing the code, I don’t see how might duplicate the annotations
```
  {{- if .Values.cloudsqlproxy.workloadIdentity.enabled }}
  annotations:
  {{- with .Values.serviceAccount.annotations  }}
      {{- toYaml . | nindent 4 }}
  {{- end }}
    {{ template "prometheus-mysql-exporter.workloadIdentityAnnotation" . }}
  {{- else }}
  {{- with .Values.serviceAccount.annotations  }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```
If WID is enabled true, the code will go on the first condition
```
  {{- if .Values.cloudsqlproxy.workloadIdentity.enabled }}
  annotations:
  {{- with .Values.serviceAccount.annotations  }}
      {{- toYaml . | nindent 4 }}
  {{- end }}
```
And if not enabled, the code will go with the else
```
  {{- else }}
  {{- with .Values.serviceAccount.annotations  }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```
I tested with different use cases and I don’t see how
Case 1: 
- serviceAccount.create:true
- serviceAccount.annotations: {}
Create the service account without annotations
Case 2: 
- serviceAccount.create:true
- serviceAccount.annotations.app: test
Create the service account with one annotation
Case 3: 
- serviceAccount.create:true
- serviceAccount.annotations:  {}
- cloudsqlproxy.enabled: true
Create the service account without annotations
Case 4: 
- serviceAccount.create:true
- serviceAccount.annotations:  {}
- cloudsqlproxy.enabled: true
- cloudsqlproxy.enabled: true
- cloudsqlproxy.workloadIdentity.enabled: true
Create the service account with one annotation
Case 5: 
- serviceAccount.create:true
- serviceAccount.annotations.app: test
- cloudsqlproxy.enabled: true
- cloudsqlproxy.enabled: true
- cloudsqlproxy.workloadIdentity.enabled: true
Create the service account with two annotation

If I'm missing some case, please let me know to correct it.

Reply:
I think it is resolved now, thanks @zanhsieh for the clarification